auth/src/main/java/org/aerogear/mobile/auth/utils/CertificatePinningCheck.java

```diff
@@ -0,0 +1,32 @@
+package org.aerogear.mobile.auth.utils;
+
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+
+
+public class CertificatePinningCheck {
+    private final HttpServiceModule httpModule;
+    private boolean isComplete = false;
+    private Exception error;
+
+    public CertificatePinningCheck(final HttpServiceModule httpModule) {
+        this.httpModule = httpModule;
+    }
+
+    public void check(final String url) {
+        HttpRequest request = httpModule.newRequest();
+        request.get(url);
+        HttpResponse response = request.execute();
+        response.onError(() -> this.error = response.getError());
+        response.onComplete(() -> this.isComplete = true);
+    }
+
+    public boolean isComplete() {
```

Comment:
IntelliJ is complaining that this method isn't used so be sure to include it in your tests.